lib/data_update_scripts/20200904132553_remove_draft_articles_with_duplicate_feed_source_url.rb

```diff
@@ -0,0 +1,78 @@
+module DataUpdateScripts
+  class RemoveDraftArticlesWithDuplicateFeedSourceUrl
+    def run
+      # Currently there are duplicate draft articles in the DB with the same feed_source_url.
+
+      # This statement deletes all draft articles in excess found to be duplicate over feed_source_url,
+      # excluding those whose body_markdown is different from the other duplicate occurrences
+      result = ActiveRecord::Base.connection.execute(
+        <<~SQL,
+          WITH duplicates_draft_articles AS
+              (SELECT id
+               FROM
+                   (SELECT id,
+                           published,
+                           body_markdown,
+                           LAG(body_markdown, 1) OVER(PARTITION BY feed_source_url
+                                                      ORDER BY id ASC) AS previous_body_markdown,
+                           ROW_NUMBER() OVER(PARTITION BY feed_source_url
+                                             ORDER BY id ASC) AS row_number
+                    FROM articles
+                    WHERE feed_source_url IS NOT NULL ) duplicates
+               WHERE duplicates.row_number > 1
+                   AND published = 'f' -- drafts
+                   AND body_markdown = previous_body_markdown -- with the same body
+           )
+          DELETE
+          FROM articles
+          WHERE id IN (SELECT id FROM duplicates_draft_articles) RETURNING id;
+        SQL
+      )
+
+      # Sending IDs of deleted articles to Datadog
+      result.map { |row| row["id"] }.in_groups_of(1000) do |ids|
+        DatadogStatsClient.event(
```

Comment:
Two notes for when we do this one
1) We need to also remove these from Elasticsearch. We can do that with the IDs we gather from this statement
2) I dont know how to view these IDs in Datadog. I got the other IDs from Elasticsearch when I went to remove those but if we cant see these in Datadog we should send them another way or somewhere else.